src/services/mail-service.js

```diff
@@ -1,149 +1,106 @@
 import nodemailer from 'nodemailer';
 import Logger from '../common/logger.js';
 import ServerConfig from '../bootstrap/server-config.js';
+import ResourceManager from '../resources/resource-manager.js';
+import { SUPPORTED_UI_LANGUAGES } from '../resources/ui-language.js';
 
 const logger = new Logger(import.meta.url);
+
 class MailService {
-  static get inject() { return [ServerConfig]; }
+  static get inject() { return [ServerConfig, ResourceManager]; }
 
-  constructor(serverConfig) {
+  constructor(serverConfig, resourceManager) {
+    this.resourceManager = resourceManager;
     this.emailSenderAddress = serverConfig.emailSenderAddress;
     this.transport = nodemailer.createTransport(serverConfig.smtpOptions);
+
+    this.translators = SUPPORTED_UI_LANGUAGES.map(language => this.resourceManager.createI18n(language).t);
   }
 
-  sendRegistrationVerificationLink({ username, email, verificationLink }) {
+  sendRegistrationVerificationEmail({ email, username, verificationLink }) {
     logger.info(`Creating email with registration verification link ${verificationLink}`);
 
-    const germanText
-      = 'Willkommen!\n\n'
-      + `Sie haben sich erfolgreich als ${username} auf ELMU registriert.\n`
-      + `Bitte bestätigen Sie Ihre Registrierung hier: ${verificationLink}`;
-
-    const germanHtml
-      = '<p>Willkommen!<br/><br/>'
-      + `Sie haben sich erfolgreich als ${username} auf ELMU registriert.<br/>`
-      + `Bitte bestätigen Sie Ihre Registrierung hier: <a href="${verificationLink}">Registrierung bestätigen</a></p>`;
-
-    const englishText
-      = 'Welcome!\n\n'
-      + `You have registered successfully with ELMU as ${username}.\n`
-      + `Please confirm your registration here: ${verificationLink}`;
-
-    const englishHtml
-      = '<p>Welcome!<br/><br/>'
-      + `You have registered successfully with ELMU as ${username}.<br/>`
-      + `Please confirm your registration here: <a href="${verificationLink}">confirm registration</a></p>`;
-
-    const message = {
-      from: this.emailSenderAddress,
-      to: email,
-      subject: 'Willkommen auf ELMU! / Welcome to ELMU!',
-      text: `${germanText}\n\n${englishText}`,
-      html: `${germanHtml}\n${englishHtml}`
-    };
+    const subject = this.translators
+      .map(t => t('mailService:registrationVerificationEmail.subject', { username, verificationLink }))
+      .join(' / ');
+
+    const text = this.translators
+      .map(t => t('mailService:registrationVerificationEmail.text', { username, verificationLink }))
+      .join('\n\n');
+
+    const html = this.translators
+      .map(t => t('mailService:registrationVerificationEmail.html', { username, verificationLink }))
```

Comment:
Actually, we should HTML encode these values. What if someone chooses this username: "<script>doSmthBad()</script>"?

Reply:
Very good point, done, thanks!